test/integration/jobschedulingpreference_test.go

```
@@ -0,0 +1,264 @@
+		clusters[1]: w2,
+	}
+
+	return jspSpec
+}
+func createJSPTestObjs(jspSpec fedschedulingv1a1.JobSchedulingPreferenceSpec, namespace string, fedClient clientset.Interface) (string, error) {
+	// TODO(irfanurrehman): should the order of this creation matter?
+	t, err := fedClient.CoreV1alpha1().FederatedJobs(namespace).Create(getFederatedJobTemplate(namespace))
+	if err != nil {
+		return "", err
+	}
+
+	jsp := &fedschedulingv1a1.JobSchedulingPreference{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      t.Name,
+			Namespace: namespace,
+		},
+		Spec: jspSpec,
+	}
+	_, err = fedClient.SchedulingV1alpha1().JobSchedulingPreferences(namespace).Create(jsp)
+	if err != nil {
+		return "", err
+	}
+
+	return t.Name, nil
+}
+
+func waitForMatchingJobPlacement(fedClient clientset.Interface, name, namespace string, expected map[string]adapters.JobSchedulingResult) error {
+	err := wait.Poll(framework.DefaultWaitInterval, wait.ForeverTestTimeout, func() (bool, error) {
+		clusterNames := []string{}
+		p, err := fedClient.CoreV1alpha1().FederatedJobPlacements(namespace).Get(name, metav1.GetOptions{})
+		clusterNames = p.Spec.ClusterNames
+		if err != nil {
+			return false, nil
+		}
+
+		if len(clusterNames) > 0 {
+			totalClusters := 0
+			for _, clusterName := range clusterNames {
+				totalClusters++
+				_, exists := expected[clusterName]
+				if !exists {
+					return false, nil
+				}
+			}
+
+			// All clusters in placement has a matched cluster name as in expected.
+			if totalClusters == len(expected) {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+
+	return err
+}
+
+func waitForMatchingJobOverride(fedClient clientset.Interface, name, namespace string, expected map[string]adapters.JobSchedulingResult) error {
+	err := wait.Poll(framework.DefaultWaitInterval, wait.ForeverTestTimeout, func() (bool, error) {
+		override, err := fedClient.CoreV1alpha1().FederatedJobOverrides(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
```

Why is it desirable to ignore this error?

---

Fixed!